airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_file_schema_loader.py

```diff
@@ -50,7 +51,7 @@ def __post_init__(self, options: Mapping[str, Any]):
             self.file_path = _default_file_path()
         self.file_path = InterpolatedString.create(self.file_path, options=options)
 
-    def get_json_schema(self) -> Mapping[str, Any]:
+    def get_schema(self) -> Mapping[str, Any]:
```

Comment:
If we are renaming `get_json_schema()` to `get_schema()` here, we will also need to rename it on the parent interface class `SchemaLoader` not just in `DefaultSchemaLoader` There are a few places where we use `SchemaLoader.get_json_schema()` at runtime so we need to properly override it so that it gets used everywhere. 

Alternatively, if we want to simplify this, we could also leave this as `get_json_schema()`, so the parent interface doesn't have to change. And from the perspective of the declarative components we just don't reuse the parent class `ResourceSchemaLoader.get_schema(self, name: str)`. And this still allows us to reuse the recursive resolver.

Reply:
@artem1205 
`JsonFileSchemaLoader` also extends `SchemaLoader` so it needs to implement `get_json_schema` too

Reply:
I've overwritten `get_json_schema` -> `get_schema` https://github.com/airbytehq/airbyte/pull/19718/files#diff-eebfee995e9a963a52d517c0ebcab9579872dee179cf961719740e6c3fdddc9fR40.
So we do not really need SchemaLoader anymore

Reply:
I don't think we should be getting rid of `SchemaLoader`. This JsonFileSchemaLoader and the DefaultSchemaLoader are intended to be subclasses of the `SchemaLoader`. From within parts of the low-code CDK, we invoke this `get_json_schema()` at runtime. For example here: https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_stream.py#L132

If we rename this method in this class, we should also rename it in `SchemaLoader` as well instead of removing the interface altogether. And then we also rename wherever it's being used like in the file I mentioned. Just renaming this method name will probably break out backend because we're using it in a few different places.

Reply:
:white_check_mark: Refactored, renamed it all back to `get_json_schema`